tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -1389,3 +1391,108 @@ def test_throttling(self, cloud_storage_type):
+        assert self.redpanda is not None
+
+    def _start_redpanda(self, override_cfg_params: Optional[dict] = None):
+        if override_cfg_params:
+            self.redpanda.add_extra_rp_conf(override_cfg_params)
+        self.redpanda.start()
+        for topic in self.topics:
+            self.kafka_tools.create_topic(topic)
+
+    def _all_kafka_connections_closed(self):
+        count = 0
+        for n in self.redpanda.nodes:
+            for family in self.redpanda.metrics(
+                    n, metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS):
+                for sample in family.samples:
+                    if sample.name == "redpanda_rpc_active_connections" and sample.labels[
+                            "redpanda_server"] == "kafka":
+                        connections = int(sample.value)
+                        self.logger.debug(
+                            f"open connections: {connections} on node {n.account.hostname}"
+                        )
+                        count += connections
+        return count == 0
+
+    def _test(self):
+        msg_count = 5 * (self.segment_size // 2048)
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       self.s3_topic_name,
+                                       msg_size=2048,
+                                       msg_count=msg_count)
+        producer.start()
+        producer.wait(timeout_sec=300)
+        producer.free()
+
+        original_snapshot = self.redpanda.storage(
+            all_nodes=True).segments_by_node("kafka", self.topic, 0)
+
+        for node, node_segments in original_snapshot.items():
+            assert len(
+                node_segments
+            ) >= 5, f"Expected at least 10 segments, but got {len(node_segments)} on {node}"
+
+        self.kafka_tools.alter_topic_config(
+            self.topic,
+            {
+                TopicSpec.PROPERTY_RETENTION_LOCAL_TARGET_BYTES:
+                self.segment_size,
+            },
+        )
+
+        wait_for_removal_of_n_segments(redpanda=self.redpanda,
+                                       topic=self.topic,
+                                       partition_idx=0,
+                                       n=2,
+                                       original_snapshot=original_snapshot)
+
+        def segments_downloaded():
+            downloads = self.redpanda.metric_sum(
+                'vectorized_cloud_storage_successful_downloads_total')
+            self.logger.debug(f'downloads: {downloads}')
+            return downloads > 0
+
+        assert not segments_downloaded()
+
+        try:
+            # The rpk process is killed after one second, due to the relatively larger segment
+            # size, this should ensure that the client is disconnected during hydrate.
+            self.rpk.consume(self.topic, timeout=1)
+        except RpkException as e:
+            assert 'timed out' in e.msg
```

Comment:
Ideally, we would somehow be able to slow down hydration request to cloud storage so much that we are able to deterministically abort rpk while hydrate is in progress. 

That doesn't seem to be easily achievable so we have to resort to setting `cloud_storage_hydration_timeout_ms` to a very low value and killing rpk process early.

Reply:
Maybe not for this PR, but in the future perhaps we could make use of `kafka_consume_transport` in a fixture test? It's not quite as end-to-end as a ducktape test, but would it be easier to reproduce what you're looking for here?

Reply:
Thanks for the pointer, I'll look it up